src/Jackalope/Node.php

```diff
@@ -1452,7 +1467,8 @@ public function unsetProperty($name)
      */
     public function confirmSaved()
     {
-        foreach ($this->properties as $property) {
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
             if ($property->isModified() || $property->isNew()) {
```

Comment:
i think you don't need loadProperty here. if it was just loaded, it will be neither new nor modified. you could loop over the already existing property objects for this.